### PR TITLE
Core to ansyn

### PR DESCRIPTION
**Pull request by yairtawil**

### CODE REVIEW

File: src/app/@ansyn/ansyn/modules/menu-items/tools/actions/tools.actions.ts [hunk 1]
```diff
@@ -33,7 +34,8 @@ export const ToolsActionsTypes = {
 	STORE: {
 		SET_ANNOTATION_MODE: type('SET_ANNOTATION_MODE')
 	},
-	SET_ACTIVE_ANNOTATION_LAYER: 'SET_ACTIVE_ANNOTATION_LAYER'
+	SET_ACTIVE_ANNOTATION_LAYER: 'SET_ACTIVE_ANNOTATION_LAYER',
+	CLEAR_ACTIVE_INTERACTIONS: 'CLEAR_ACTIVE_INTERACTIONS'
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/menu-items/tools/actions/tools.actions.ts]: what this action do?

File: src/app/@ansyn/imagery/public_api.ts [hunk 2]
```diff
@@ -1,3 +1,48 @@
+export { IOverlaysCriteriaOptions } from './model/overlay.model';
+
+export { GeoRegisteration } from './model/overlay.model';
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/public_api.ts]: 🆘

File: src/app/@ansyn/map-facade/services/geocoder.service.ts [hunk 3]
```diff
@@ -23,7 +21,7 @@ export class GeocoderService {
 			map(res => res.resourceSets[0].resources[0]),
 			map(res => res ? { ...res.point, coordinates: res.point.coordinates.reverse() } : null),
 			catchError((error: Response | any) => {
-				this.errorHandlerService.httpErrorHandle(error);
+				// this.errorHandlerService.httpErrorHandle(error);
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/services/geocoder.service.ts]: add todo?

File: src/app/@ansyn/map-facade/test/async-observable-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// From https://angular.io/guide/testing
+
+import { defer, Observable } from 'rxjs';
+
+/** Create a truly async observable that emits-once and completes
+ *  after a JS engine turn */
+export function asyncData<T>(data: T): Observable<T> {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/test/async-observable-helpers.ts]: explain

File: src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts [hunk 5]
```diff
@@ -95,11 +95,11 @@ export class SelectCaseAppEffects {
 			new UpdateOverlaysManualProcessArgs({ override: true, data: overlaysManualProcessArgs }),
 			new UpdateFacetsAction(facets),
 			new UpdateSelectedLayersIds(activeLayersIds),
-			new SetContextParamsAction({ contextEntities }),
+			<any> { type: '[Context] Set context params', payload: { contextEntities }},
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/ansyn/app-effects/effects/cases/select-case.app.effects.ts]: todo

File: src/app/@ansyn/ansyn/app-effects/effects/cases/update-case.app.effects.ts [hunk 6]
```diff
@@ -1,30 +1,25 @@
...
+import { ICase } from '@ansyn/imagery';
+import { selectSelectedLayersIds } from '../../../modules/menu-items/layers-manager/reducers/layers.reducer';
+import { selectFacets } from '../../../modules/menu-items/filters/reducer/filters.reducer';
+import { selectComboBoxesProperties } from '../../../modules/status-bar/reducers/status-bar.reducer';
+import { selectOverlaysManualProcessArgs } from '../../../modules/menu-items/tools/reducers/tools.reducer';
+import { UpdateCaseAction } from '../../../modules/menu-items/cases/actions/cases.actions';
+import { selectAutoSave, selectSelectedCase } from '../../../modules/menu-items/cases/reducers/cases.reducer';
+
+// HACK //
+const selectContextEntities: any = createSelector(createFeatureSelector('context'), (context: any) => context.entities);
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/ansyn/app-effects/effects/cases/update-case.app.effects.ts]: todo

File: src/app/@ansyn/ansyn/app-effects/effects/map.app.effects.spec.ts [hunk 7]
```diff
@@ -13,37 +13,16 @@ import {
...
-import {
-	IStatusBarState,
-	statusBarFeatureKey,
-	StatusBarInitialState,
-	StatusBarReducer,
-	statusBarStateSelector
-} from '../../modules/status-bar/public_api';
-import * as extentFromGeojson from '@ansyn/core';
+} from '@ansyn/map-facade';
+import * as extentFromGeojson from '@ansyn/map-facade';
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/ansyn/app-effects/effects/map.app.effects.spec.ts]: spyOn

File: src/app/@ansyn/ansyn/modules/core/actions/core.actions.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
+// import { Action } from '@ngrx/store';
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/ansyn/modules/core/actions/core.actions.ts]: 👍

File: src/app/@ansyn/ansyn/modules/menu-items/tools/actions/tools.actions.ts (hunk 1 above)
Comment (yairtawil) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/actions/tools.actions.ts]: CLEAR_ACTIVE_TOOLS

File: src/app/@ansyn/imagery/public_api.ts (hunk 2 above)
Comment (yairtawil) [reply src/app/@ansyn/imagery/public_api.ts]: Extends from outside

File: src/app/@ansyn/map-facade/test/async-observable-helpers.ts (hunk 4 above)
Comment (yairtawil) [reply src/app/@ansyn/map-facade/test/async-observable-helpers.ts]: ram

File: src/app/@ansyn/map-facade/services/geocoder.service.ts (hunk 3 above)
Comment (yairtawil) [reply src/app/@ansyn/map-facade/services/geocoder.service.ts]: toast

**Pull request merged**